docs/kbs_protocol.md

```diff
@@ -0,0 +1,265 @@
+# KBS Protocol
+
+The Key Broker protocol defines communication between a Key Broker Client in a confidential guest 
+and a trusted Key Broker Service. The protocol uses the simple, universal, and extensible 
+"Request-Challenge-Attestation-Response" (RCAR) method to facilitate guest attestation and secret injection.
+
+# Introduction
+
+The purpose of the attestation between KBS and KBC is actually to confirm whether 
+**the platform where the KBC is located is in the expected security state**, that is, it runs in 
+a real and harmless HW-TEE.
+
+In Universal-KBS, HW-TEE attestation process is the semantics of the application 
+layer, which is defined as a simple, universal and extensible 
+"Request-Challenge-Attestation-Response" (RACR) protocol. The temporary asymmetric key generated 
+by HW-TEE is used to encrypt the response payload, and the token mechanism is used to avoid the 
+performance problems caused by multiple attestation.
+
+In order to ensure the ease of use and security completeness of Universal-KBS, we will use HTTPS 
+as the default transmission method to carry the application layer semantics designed in this 
+document. This is because HTTPS provides KBC with a means to authenticate KBS identity, which 
+effectively avoids malicious attackers from hijacking KBS address to impersonate KBS and deceive KBC.
+In order to achieve this, the public key of KBS needs to be transmitted to KBC through an effective way, 
+and the specific way of public key distribution is not within the scope of this document.
+In addition, it should be noted that the confidentiality protection provided by HTTPS alone is not enough 
+to meet the security requirements of KBS protocol, therefore, as mentioned above, the key generated 
+by hw-tee needs to be used to encrypt and protect the confidential data.
+
+# RCAR semantics
+
+The semantics of attestation defined by Universal-KBS is a simple and extensible four-step model, 
+which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC actively sends to KBS, calls the service API provided by KBS to request resources or services.
+2. **Challenge**: After receiving the request, KBS returns a challenge to KBC and asks KBC to 
```

Comment:
> I guess the only real reason for `Challenge` is to provide a nonce?

One of the important reason to add this `Challenge` is to add a nonce to protect the protocol from the replay attack during attestation.
But as this document's `Challenge` mentioned, it also inlcudes this `extra-params` section. Because we think this Four Steps negotation protocol should also compatible with the types of HW-TEE (like SEV-ES) who relies on Diffie–Hellman to build a security channel between the KBC and KBS firstly to transport the evidence and sensitive data.
So this protocol's step1 and step2's `extra-params` can used to transfer these essential data to finish the Diffie–Hellman. Then the following step3 and step4 can be transported via this security channel.